packages/components-vue/projects/vue-wrapper/src/plugin.ts

```diff
@@ -0,0 +1,45 @@
+// plugin
+import { App, ref, inject } from 'vue';
+
+import { load, componentsReady } from '@porsche-design-system/components-js';
+import { prefixInjectionKey } from './utils';
+
+export type PorscheDesignSystemPluginOptions = {
+  prefix?: string;
+  extends?: Record<string, unknown>;
+}
+
+export interface PorscheDesignSystemPlugin {
+  [key: string]: any;
+  install: (app: App, options?: PorscheDesignSystemPluginOptions) => void;
+}
+
+export const porscheDesignSystemSymbol = Symbol();
+
+export function usePorscheDesignSystemPlugin() {
+  const porscheDesignSystem = inject(porscheDesignSystemSymbol);
+  if (!porscheDesignSystem) throw new Error('No PorscheDesignSystem provided!!!');
+
+  return porscheDesignSystem;
+}
+
+export function createPorscheDesignSystem(options = { prefix: '' }) {
+  const isPorscheDesignSystemLoaded = ref(false);
+
+  const $porscheDesignSystem: PorscheDesignSystemPlugin = {
+    options,
+    isPorscheDesignSystemLoaded,
+    componentsReady,
+    async install(app) {
+      load();
+
+      await componentsReady();
```

Comment:
@alvarosabu Is it really needed to await the components readiness here? Also, If you set the `isPorscheDesignSystemLoaded.value` to `true` after the `await`, it might lead to multiple initializations in setups where you load multiple apps on one page. We at Canvas will have that when we do the Astro Jamstack pages.

Reply:
@leifmarcus For the use with Astro wouldn't make more sense to use the ProviderWrapper instead?

Reply:
Regarding the await, isPorscheDesignSystemLoaded.value is true for multiple instance, but we needed to know when we can render the components. I can only set it if it was false first to avoid what you saying